src/mlpack/methods/kmeans/elkan_kmeans_impl.hpp

```diff
@@ -149,13 +176,23 @@ double ElkanKMeans<DistanceType, MatType>::Iterate(const arma::mat& centroids,
     // At this point, we know the new cluster assignment.
     // Step 4: for each center c, let m(c) be the mean of the points assigned to
     // c.
-    newCentroids.col(assignments[i]) += arma::vec(dataset.col(i));
-    counts[assignments[i]]++;
+    threadNewCentroids[threadId].col(assignments[i]) += arma::vec(dataset.col(i));
+    threadCounts[threadId][assignments[i]]++;
+  }
+
+  // Combine thread-local results
+  for (int t = 0; t < numThreads; ++t)
+  {
+    newCentroids += threadNewCentroids[t];
+    counts += threadCounts[t];
   }
 
   // Now, normalize and calculate the distance each cluster has moved.
   arma::vec moveDistances(centroids.n_cols);
   double cNorm = 0.0; // Cluster movement for residual.
+  #ifdef MLPACK_USE_OPENMP
```

Comment:
No need for the ifdef; the compiler will ignore the pragma if OpenMP is not enabled. :+1: